qutip/solver/scattering.py

```diff
@@ -109,53 +47,74 @@ def set_partition(collection, num_sets):
         yield tuple(tuple(indices) for indices in partition)
 
 
-def photon_scattering_operator(evolver, c_ops, taus_list):
+def photon_scattering_amplitude(evolver, c_ops, tlist, indices, psi, psit):
     """
-    Compute the scattering operator for a system emitting into multiple
+    Compute the scattering amplitude for a system emitting into multiple
     waveguides.
 
     Parameters
     ----------
-    evolver : :class: qutip.scattering.Evolver
-        Evolver-wrapped Hamiltonian describing the system.
+    evolver : :class:Propagator
+        Propagator
     c_ops : list
         list of collapse operators for each waveguide; these are assumed to
         include spontaneous decay rates, e.g.
         :math:`\\sigma = \\sqrt \\gamma \\cdot a`
-    taus_list : list-like
-        List of (list of emission times) for each waveguide.
-
-    Returns
-    -------
-    omega : :class: qutip.Qobj
-        The temporal scattering operator with dimensionality equal to the
-        system state.
+    tlist : array_like
+        Times at which emissions can happen.
+    indices : list-like
+        List of (list of emission times indices) for each waveguide.
```

Comment:
I think the list of emission times for each waveguide was a better interface than these lists of indices into tlist. It's pretty easy to construct the tlist and the list of indices insides this function if we need them, but very annoying for the user to have to construct the tlist and the list of indices themselves every time they would like to calculate the scattering amplitudes.